features/steps/component.go

```diff
@@ -64,16 +67,38 @@ func NewComponent() *Component {
 	}
 
 	c.serviceList = service.NewServiceList(initMock)
-	c.S3UploadedSitemap = make(map[config.Language]string)
+
+	c.files = make(map[string]string)
+	c.S3UploadedSitemap = make(map[string]string)
+	c.welshVersion = make(map[string]bool)
+
 	return c
 }
 
 func (c *Component) Close() {
-	os.Remove(c.cfg.RobotsFilePath)
+	c.CleanFile(c.cfg.RobotsFilePath)
```

Comment:
Both the Close() and Reset() functions are doing exactly the same things. So only one of them is needed. Perhaps delete the Close() function and replace any usages with calls to Reset() instead. My understanding is that the Close() function would normally be used to close the HttpServer but there isn't one to close here so we don't need it.